src/vs/server/node/auth-http.ts

```diff
@@ -0,0 +1,70 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import * as http from 'http';
+import { ILogService } from 'vs/platform/log/common/log';
+import { parse } from 'querystring';
+import { args } from 'vs/server/node/args';
+import { authenticated, generateAndSetPassword, handlePasswordValidation } from 'vs/server/node/auth';
+import { APP_ROOT, serveError, serveFile } from 'vs/server/node/server.main';
+
+const LOGIN = path.join(APP_ROOT, 'out', 'vs', 'server', 'browser', 'workbench', 'login.html');
+
+export async function handleServerRequest(pathname: string | null, req: http.IncomingMessage, res: http.ServerResponse, logService: ILogService): Promise<boolean> {
+	if (args.password === undefined) {
+		await generateAndSetPassword(logService);
+	}
+	const auth = await authenticated(args, req);
+	if (!auth) {
+		if (pathname === '/') {
+			serveFile(logService, req, res, LOGIN);
+			return true;
+		}
+		if (pathname === '/login') {
+			const password = (await collectRequestData(req)).password;
+			const { isPasswordValid, hashedPassword } = await handlePasswordValidation({
+				passwordFromRequestBody: password,
+				passwordFromArgs: args.password,
+			});
+
+			if (isPasswordValid) {
+				res.writeHead(302, {
+					'Location': '/',
+					'Set-Cookie': `key=${hashedPassword}`,
```

Comment:
please add `HttpOnly` attribute

I'm not actually sure that auth with cookies work like that, i.e. caching the password in the client browser. Usually password is send only once and then session cookie is generated which is stored in server and set as a cookie to cache in the browser?